### PR TITLE
Add URL to error/warning messages

### PR DESCRIPTION
**Pull request by jdidion**

### CODE REVIEW

File: src/main/scala/wdlTools/syntax/package.scala [hunk 1]
```diff
@@ -64,8 +64,8 @@ object TextSource {
 
 // Syntax error exception
 final class SyntaxException(message: String) extends Exception(message) {
-  def this(msg: String, text: TextSource) =
-    this(s"${msg} at ${text}")
+  def this(msg: String, text: TextSource, docSourceURL: Option[URL] = None) =
+    this(s"${msg} at ${text}${docSourceURL.map(url => s" in ${url.toString}").getOrElse("")}")
```
Comment (orodeh) [inline src/main/scala/wdlTools/syntax/package.scala:68]: Yes, it is fun to write this in one line. However, for readability, can you please break this into two lines?

File: src/main/scala/wdlTools/typing/package.scala [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
+import java.net.URL
+
 import wdlTools.syntax.TextSource
 
 // Type error exception
 final class TypeException(message: String) extends Exception(message) {
-  def this(msg: String, text: TextSource) =
-    this(s"${msg} at ${text}")
+  def this(msg: String, text: TextSource, docSourceURL: Option[URL] = None) =
+    this(s"${msg} at ${text}${docSourceURL.map(url => s" in ${url.toString}").getOrElse("")}")
```
Comment (orodeh) [inline src/main/scala/wdlTools/typing/package.scala:10]: Can you please calculate the document string in a separate line?

File: src/main/scala/wdlTools/syntax/package.scala (hunk 1 above)
Comment (jdidion) [reply src/main/scala/wdlTools/syntax/package.scala:68]: I tried but it doesn't seem to work - the compiler complains about "this does not accept arguments"

Comment (jdidion) [reply src/main/scala/wdlTools/syntax/package.scala:68]: It looks like Java requires that if you make a call to this() in the constructor it has to be the first line.

**Comment by orodeh** [comment]
Alright, merge it then, and I will take a look at it when I get a chance.

File: src/main/scala/wdlTools/syntax/package.scala (hunk 1 above)
Comment (jdidion) [reply src/main/scala/wdlTools/syntax/package.scala]: I added a separate formatMessage function that I call from the constructor.

File: src/main/scala/wdlTools/typing/package.scala (hunk 2 above)
Comment (jdidion) [reply src/main/scala/wdlTools/typing/package.scala]: Same as above

**Pull request merged**